fix: Or filters with warnings

danieljbruce opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Should be merged after or-queries PR. Adds warnings when `.filter` is not used correctly and modifies tests to react to warnings correctly.

danieljbruce committed:
or filters interface

danieljbruce committed:
implementation of new filter

danieljbruce committed:
Remove duplicate code

danieljbruce committed:
Parse with type guard

danieljbruce committed:
Remove the newFilter variable

danieljbruce committed:
Add filter and enable chaining

danieljbruce committed:
test add filter

danieljbruce committed:
Add another unit test

danieljbruce committed:
Add system-test stubs

danieljbruce committed:
Add system-tests for the OR filter

danieljbruce committed:
Move things around

danieljbruce committed:
Merge branch 'main' into or-filters

danieljbruce committed:
Added a unit test

danieljbruce committed:
Merge branch 'or-filters' of https://github.com/danieljbruce/nodejs-datastore into or-filters

danieljbruce committed:
Add a unit test for OR filters

danieljbruce committed:
Just use filter method

danieljbruce committed:
new warning test

danieljbruce committed:
Revert "new warning test"

This reverts commit 37400a6d3a9d7385a0203db70949f037e4c35de4.

danieljbruce committed:
Now removes deprecation warning properly

danieljbruce committed:
Add a test for new warning

danieljbruce committed:
Add setAncestor

Adds a new setAncestor method for ensuring only one ancestor is set for the query at a time. This will avoid errors that result because of setting multiple ancestors. Also deprecate hasAncestor because it will lead to warnings like this. Add parser logic to use the value provided in setAncestor for query sent to backend.

danieljbruce committed:
Basic unit tests for setAncestor

Added tests for the query proto, one to make sure the query structure is right when setting ancestor once and one to make sure the query structure is right when setting ancestor twice. Also added a unit test to make sure that ancestor is set the right way internally when using setAncestor.

danieljbruce committed:
change expected result for OR query

This code change adjusts the expected result for running an OR query. Old result used to correspond with AND, but now corresponds to OR.

danieljbruce committed:
Fix a test by not requiring the done callback

A test is timing out because we are waiting for done to be called. This fix does not require done to be called.

danieljbruce committed:
Merge branch 'main' into or-filters

danieljbruce committed:
Merge branch 'main' of https://github.com/googleapis/nodejs-datastore into or-filters

# Conflicts:
#	src/query.ts
#	test/query.ts

danieljbruce committed:
Revert "Fix a test by not requiring the done callback"

This reverts commit 1159b374f483d107736a765b5b4f1ade80116ce3.

danieljbruce committed:
Revert "Basic unit tests for setAncestor"

This reverts commit 86841d6e0a3d26e2df647fa4a42ed849b61b38a6.

danieljbruce committed:
Revert "Add setAncestor"

This reverts commit e84582faa6041429a4e81d47bd21dbc338a639e7.

danieljbruce committed:
Separate filters and new filters internally

This commit is done to avoid a breaking typescript change which could have the potential to affect some users who read `filters` on a query as its type had been changed to be more flexible, but is now back to what it was.

danieljbruce committed:
Move AND/OR into their own separate function

AND and OR should not be static functions of the filter class because then the user has to type Filter.AND instead of AND for example.

danieljbruce committed:
Eliminate unused imports

Artifacts of having imports laying around and moving functionality between files

danieljbruce committed:
Revert "Add a test for new warning"

This reverts commit bc15f325d87b4ffe46ae8727293c5cefa2300501.

danieljbruce committed:
Revert "Now removes deprecation warning properly"

This reverts commit db02a5089242b25b4bebae4304a4778f0bf6d17f.

danieljbruce committed:
Revert "Revert "Now removes deprecation warning properly""

This reverts commit 8e96bf5e5ef11593853ae061fd1858c2e463d6ce.

danieljbruce committed:
Revert "Revert "Add a test for new warning""

This reverts commit 1ddeea6a75d01ad3c24e46d85e31eca73235790a.

danieljbruce committed:
Merge branch 'main' of https://github.com/googleapis/nodejs-datastore into or-filters-with-warnings

# Conflicts:
#	src/entity.ts
#	src/filter.ts
#	src/query.ts
#	system-test/datastore.ts
#	test/entity.ts
#	test/query.ts

danieljbruce committed:
fix: Or filters with warnings

kolea2 commented on src/query.ts:
```diff
@@ -218,6 +218,9 @@ class Query {
       return this;
     } else {
+      process.emitWarning(
+        'Providing Filter objects is recommended when using .filter'
```
nit - mention an example like a CompositeFilter

kolea2 reviewed (approved):
lgtm after nit

kolea2 commented:
also, please rename PR with conventional commit prefix

danieljbruce committed:
Merge branch 'or-filters-with-warnings' of https://github.com/danieljbruce/nodejs-datastore into or-filters-with-warnings

danieljbruce committed:
Change the warning message to include examples

Change the warning message to include examples as per the nit mentioned in the PR.
